ogr/ogrsf_frmts/hana/ogrhanalayer.cpp

```diff
@@ -765,41 +765,69 @@ OGRErr OGRHanaLayer::InitFeatureDefinition(const CPLString &schemaName,
 /*                         ReadGeometryExtent()                         */
 /************************************************************************/
 
-void OGRHanaLayer::ReadGeometryExtent(int geomField, OGREnvelope *extent)
+void OGRHanaLayer::ReadGeometryExtent(int geomField, OGREnvelope *extent,
+                                      int force, bool fastExtentMethod)
 {
     EnsureInitialized();
 
     OGRGeomFieldDefn *geomFieldDef = featureDefn_->GetGeomFieldDefn(geomField);
+    if (geomFieldDef == nullptr)
+        return;
+
     const char *clmName = geomFieldDef->GetNameRef();
     int srid = GetGeometryColumnSrid(geomField);
     CPLString sql;
-    if (dataSource_->IsSrsRoundEarth(srid))
+    if (!force && fastExtentMethod)
```

Comment:
Note, we can use fast extent estimation if it is a OGRHanaTableLayer.

Reply:
I rewrote the function to do that, looks more clean now.